doc/api/n-api.md

```diff
@@ -3229,6 +3239,13 @@ This API allocates a work object that is used to execute logic asynchronously.
 It should be freed using [`napi_delete_async_work`][] once the work is no longer
 required.
 
+`async_resource_name` should be a null-terminated, UTF-8-encoded string.
+
+*Note*: The `async_resource_name` identifier is provided by the user and should
+be representative of the type of async work being performed. It is also
+recommended to apply namespacing to the identifier, e.g. by including the
+module name.
```

Comment:
Link to [`type`](https://nodejs.org/api/async_hooks.html#async_hooks_type) in async\_hooks documentation?